internal/ir/xds.go

```diff
@@ -475,6 +462,56 @@ type HTTPRoute struct {
 	TCPKeepalive *TCPKeepalive `json:"tcpKeepalive,omitempty" yaml:"tcpKeepalive,omitempty"`
 	// Retry settings
 	Retry *Retry `json:"retry,omitempty" yaml:"retry,omitempty"`
+	// Security configuration for the route.
+	Security *Security `json:"security,omitempty" yaml:"security,omitempty"`
+}
+
+// Security holds the Security configuration associated with  the HTTP Route
+// +k8s:deepcopy-gen=true
+type Security struct {
```

Comment:
Should we call this struct `Security`, `SecurityFeature`, or `SecurityPolicy`? I'm slightly inclined to `Security`.

Reply:
hi @zhaohuabing, this work is similar to my recent PR https://github.com/envoyproxy/gateway/pull/2926/files#diff-b0330312d3f47f14bcdc8e1b589d84001a1969564c86ff59347bd56795db7ccaR451

think it through again, `SecurityFeatures` makes much sense to me. and also we can make this field inline, that would aviod so many nil check.

similar to [here](https://github.com/envoyproxy/gateway/pull/2926/files#diff-b0330312d3f47f14bcdc8e1b589d84001a1969564c86ff59347bd56795db7ccaR479), we can perform any check directly on this field